openwpm/config.py

```diff
@@ -0,0 +1,263 @@
+import os
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from dataclasses_json import dataclass_json
+
+from .errors import ConfigError
+
+BOOL_TYPE_VALIDATION_LIST = [True, False]
+DISPLAY_MODE_VALIDATION_LIST = ["native", "headless", "xvfb"]
+SUPPORTED_BROWSER_LIST = [
+    "firefox"
+]  # Using List instead of a str type to future proof the logic as OpenWPM may add support for more browsers in future
+TP_COOKIES_OPTIONALS_LIST = ["always", "never", "from_visited"]
+DB_EXTENSION_TYPE_LIST = [".db", ".sqlite"]
+LOG_EXTENSION_TYPE_LIST = [".log"]
+CONFIG_ERROR_STRING = (
+    "Found {value} as value for {parameter_name} in BrowserParams. "
+    "Supported values are {value_list}. Please look at "
+    "docs/Configuration.md#browser-configuration-options for more information"
+)
+EXTENSION_ERROR_STRING = (
+    "Found {extension} extension for {parameter_name} in ManagerParams "
+    "supported extensions are {value_list}. Please look at "
+    "docs/Configuration.md#platform-configuration-options for more information"
+)
+GENERAL_ERROR_STRING = (
+    "Found invalid value `{value}` for {parameter_name} in {params_type}. "
+    "Please look at docs/Configuration.md for more information"
+)
+OUTPUT_FORMAT_LIST = ["local", "s3"]
+
+ALL_RESOURCE_TYPES = {
+    "beacon",
+    "csp_report",
+    "font",
+    "image",
+    "imageset",
+    "main_frame",
+    "media",
+    "object",
+    "object_subrequest",
+    "ping",
+    "script",
+    "stylesheet",
+    "sub_frame",
+    "web_manifest",
+    "websocket",
+    "xbl",
+    "xml_dtd",
+    "xmlhttprequest",
+    "xslt",
+    "other",
+}
+
+
+@dataclass_json
+@dataclass
+class BrowserParams:
+    extension_enabled: bool = True
+    cookie_instrument: bool = True
+    js_instrument: bool = False
+    js_instrument_settings: list = field(
+        default_factory=lambda: ["collection_fingerprinting"]
+    )
+    http_instrument: bool = False
+    navigation_instrument: bool = False
+    save_content: bool = False
+    callstack_instrument: bool = False
+    dns_instrument: bool = False
+    seed_tar: str = None
+    display_mode: str = "native"
+    browser: str = "firefox"
+    prefs: dict = field(default_factory=dict)
+    tp_cookies: str = "always"
+    bot_mitigation: bool = False
+    profile_archive_dir: str = None
+    recovery_tar: str = None
+    donottrack: str = False
+    tracking_protection: bool = False
+
+
+@dataclass_json
+@dataclass
+class ManagerParams:
+    data_directory: str = "~/openwpm/"
+    log_directory: str = "~/openwpm/"
+    screenshot_path: str = None
+    source_dump_path: str = None
+    output_format: str = "local"
+    database_name: str = "crawl-data.sqlite"
+    log_file: str = "openwpm.log"
+    failure_limit: int = None
```

Comment:
```suggestion
    failure_limit: Optional[int] = None
```
Since we allow `failure_limit` to be `None` we should mark it as optional. Otherwise the typechecker will complain.
You'll need to import Optional from typing.